### PR TITLE
platform: add Radxa ROCK 3B platform support

### PR DESCRIPTION
**Pull request by nascs**
Signed-off-by: Nascs <nascs@radxa.com>

### CODE REVIEW

**Comment by nascs** [comment]
@RadxaYuntian  要先同步 eclipse/mraa

File: docs/radxa_rock_3b.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Radxa ROCK 3 Model C {#_Radxa}
```
Comment (RadxaYuntian) [inline docs/radxa_rock_3b.md]:
```suggestion
Radxa ROCK 3 Model B {#_Radxa}
```

File: src/arm/radxa_rock_3b.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    mraa_radxa_rock_3b_pininfo(b, 0, -1, -1, (mraa_pincapabilities_t){0,0,0,0,0,0,0,0}, "INVALID");
+    mraa_radxa_rock_3b_pininfo(b, 1, -1, -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "3V3");
+    mraa_radxa_rock_3b_pininfo(b, 2, -1, -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "5V");
+    mraa_radxa_rock_3b_pininfo(b, 3, 1, 0, (mraa_pincapabilities_t){1,1,0,0, 0,1,0,1}, "GPIO1_A0"); // can't control
+    mraa_radxa_rock_3b_pininfo(b, 4, -1, -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "5V");
+    mraa_radxa_rock_3b_pininfo(b, 5, 1, 1, (mraa_pincapabilities_t){1,1,0,0, 0,1,0,1}, "GPIO1_A1"); // can't control
+    mraa_radxa_rock_3b_pininfo(b, 6, -1, -1, (mraa_pincapabilities_t){1,0,0,0, 0,0,0,0}, "GND");
+    mraa_radxa_rock_3b_pininfo(b, 7, 0, 13, (mraa_pincapabilities_t){1,1,1,0, 0,1,0,0}, "GPIO0_B5");
+    mraa_radxa_rock_3b_pininfo(b, 8, 0, 25, (mraa_pincapabilities_t){1,0,0,0, 0,0,0,1}, "GPIO0_D1"); // Used by fiq_debugger
```
Comment (RadxaYuntian) [inline src/arm/radxa_rock_3b.c]: https://github.com/RadxaYuntian/mraa/pull/4#discussion_r1349830579

File: src/arm/radxa_rock_3b.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    b->pins[23].pwm.mux_total = 0;
+    b->pins[13].pwm.parent_id = 15; // pwm15-m0
+    b->pins[13].pwm.mux_total = 0;
+    b->pins[19].pwm.parent_id = 15; // pwm15-m1
+    b->pins[19].pwm.mux_total = 0;
+
+    mraa_radxa_rock_3b_pininfo(b, 0, -1, -1, (mraa_pincapabilities_t){0,0,0,0,0,0,0,0}, "INVALID");
+    mraa_radxa_rock_3b_pininfo(b, 1, -1, -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "3V3");
+    mraa_radxa_rock_3b_pininfo(b, 2, -1, -1, (mraa_pincapabilities_t){1,0,0,0,0,0,0,0}, "5V");
+    mraa_radxa_rock_3b_pininfo(b, 3, 1, 0, (mraa_pincapabilities_t){1,1,0,0, 0,1,0,1}, "GPIO1_A0"); // can't control
```
Comment (RadxaYuntian) [inline src/arm/radxa_rock_3b.c]: https://github.com/RadxaYuntian/mraa/pull/4#discussion_r1349830579

**Review by RadxaYuntian — CHANGES_REQUESTED** [review]

**Review by RadxaYuntian — APPROVED** [review]

**Pull request closed without merging**